File: PDOO-Civitas/CivitasR/lib/casilla_calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Casilla_calle < Casilla
    
    attr_reader :nombre, :tituloPropiedad
    
    def initialize(titulo)
      
      super(titulo.nombre)
      @tituloPropiedad = titulo
      
    end
    
    def recibe_jugador(iactual, todos)
     
      if(jugador_correcto(iactual,todos))
         informe(iactual,todos)
         nuevo = todos.at(iactual)
        
        if(!@tituloPropiedad.tiene_propietario)
          nuevo.puede_comprar_casilla
        else
          @tituloPropiedad.tramitar_alquiler(nuevo)
        end
      end
    end
    
    public
    def to_s
      casilla="\n"
        if(@nombre!=nil)
          casilla = casilla + "\n  Nombre casilla: "+@nombre.to_s
        end
            
        casilla = casilla +"\n  Tipo Casilla: CALLE"

        if(@tituloPropiedad!=nil)
            casilla = casilla + @tituloPropiedad.to_s   
        end
        return casilla
    end
  end
end

File: PDOO-Civitas/CivitasJ/src/civitas/CasillaCalle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ramonjtorres
 */
public class CasillaCalle extends Casilla{
    
    private String nombre;
    TituloPropiedad tituloPropiedad;
    
    
    CasillaCalle(TituloPropiedad titulo){
        super(titulo.getNombre());
        this.tituloPropiedad = titulo;
        this.informacion = "Es una propiedad";
    }
    
    TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos){
    
        if(jugadorCorrecto(iactual,todos)){
            this.informe(iactual, todos);
            Jugador nuevo = todos.get(iactual);
            if(!this.tituloPropiedad.tienePropietario()){
                nuevo.puedeComprarCasilla();
            }
            else this.tituloPropiedad.tramitarAlquiler(nuevo);  
        }
    }
    
    @Override
    public String toString(){
        
        String Casilla="\n";
        
        if(nombre!=null)
            Casilla = "\n  Nombre casilla: "+nombre;
        
        Casilla = Casilla +"\n  Tipo Casilla: CALLE";
        
        if(tituloPropiedad!=null)
            Casilla = Casilla + tituloPropiedad.toString();
        
        return Casilla;
    }
    
}
